pkg/types/openstack/validation/platform.go

```diff
@@ -21,6 +22,22 @@ func ValidatePlatform(p *openstack.Platform, n *types.Networking, fldPath *field
 	} else if !isValidValue(p.Cloud, validClouds) {
 		allErrs = append(allErrs, field.NotSupported(fldPath.Child("cloud"), p.Cloud, validClouds))
 	} else {
+		if p.MachinesSubnet != "" {
+			if len(p.ExternalDNS) > 0 {
+				allErrs = append(allErrs, field.InternalError(fldPath.Child("machinesSubnet"), errors.New("externalDNS cannot be set on custom subnets")))
```

Comment:
Why the InternalError ? There is Invalid too in the package..

Reply:
Or in supported too

Reply:
nit: externalDNS was specified, externalDNS is not supported with custom subnets.